x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/BytesRefState.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.ObjectArray;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BytesRefBlock;
+import org.elasticsearch.compute.data.BytesRefVector;
+import org.elasticsearch.compute.data.ConstantBytesRefVector;
+import org.elasticsearch.compute.data.ConstantNullBlock;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.compute.operator.DriverContext;
+
+public final class BytesRefState {
+
+    public enum Operations {
+        MAX, MIN
+    }
+
+     static class SingleState implements AggregatorState {
+        private BytesRef value;
+        private final Operations op;
+
+        SingleState(Operations op) {
+            this.value = null;
+            this.op = op;
+        }
+
+        public BytesRef value() { return this.value; }
+
+        @Override
+        public void close() {}
+
+        @Override
+        public void toIntermediate(Block[] blocks, int offset) {
+            assert blocks.length >= offset + 1;
+            if (this.value == null) {
+                blocks[offset] = Block.constantNullBlock(1);
+                return;
+            }
+            blocks[offset] = new ConstantBytesRefVector(this.value, 1).asBlock();
```

Comment:
I wanted to make a comment about using the block factory to construct blocks, but then I see that there is no access to it from here. Once your repo is sync'ed up it'll be easy to use the block factory, since this method will be passed a `DriverContext` (which contains the block factory)